src/main/java/gyro/aws/codebuild/ProjectResource.java

```diff
@@ -0,0 +1,619 @@
+/*
+ * Copyright 2020, Brightspot.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.aws.codebuild;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.psddev.dari.util.ObjectUtils;
+import gyro.aws.AwsResource;
+import gyro.aws.Copyable;
+import gyro.aws.iam.RoleResource;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Max;
+import gyro.core.validation.Min;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidationError;
+import software.amazon.awssdk.services.codebuild.CodeBuildClient;
+import software.amazon.awssdk.services.codebuild.model.BatchGetProjectsResponse;
+import software.amazon.awssdk.services.codebuild.model.CreateProjectResponse;
+import software.amazon.awssdk.services.codebuild.model.Project;
+import software.amazon.awssdk.services.codebuild.model.ProjectArtifacts;
+import software.amazon.awssdk.services.codebuild.model.ProjectFileSystemLocation;
+import software.amazon.awssdk.services.codebuild.model.ProjectSource;
+import software.amazon.awssdk.services.codebuild.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.codebuild.model.Tag;
+
+@Type("project")
+public class ProjectResource extends AwsResource implements Copyable<BatchGetProjectsResponse> {
```

Comment:
The model should not be a response, it should be the singular model that the gyro resource is modeling

```suggestion
public class ProjectResource extends AwsResource implements Copyable<Project> {
```